src/http/client.rs

```diff
@@ -1319,6 +1319,52 @@ impl Http {
         .await
     }
 
+    /// Edits a message and its attachments by Id.
+    ///
+    /// **Note**: Only the author of a message can modify it.
+    pub async fn edit_message_and_attachments(
+        &self,
+        channel_id: u64,
+        message_id: u64,
+        map: &Value,
+        new_attachments: impl IntoIterator<Item = AttachmentType<'_>>,
+    ) -> Result<Message> {
+        // TODO: Avoid duplication with send_files()?
```

Comment:
The code of `send_files` and this method is remarkably similar, but I would personally leave this be for now. I would attempt to de-duplicate code once it's used three times or more, as far as I can tell, this is just the second time.

Reply:
Yeah, it seems that's a [common role of thumb](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) in programming